tecken/api/views.py

```diff
@@ -628,3 +631,69 @@ def nones_to_zero(obj):
 
     context = {"stats": numbers}
     return http.JsonResponse(context)
+
+
+# Store a result for 10 minutes
+SYMINFO_RESULT_TIMEOUT = 600
+
+# Indicates there's nothing in the cache
+NO_VALUE_IN_CACHE = object()
+
+
+@metrics.timer_decorator("lookup_by_syminfo.timing")
+def cached_lookup_by_syminfo(somefile, someid, refresh_cache=False):
+    """Looks up somefile/someid in fileupload data; caches result
+
+    This value is cached.
+
+    :arg somefile: a string that's either a debug_file or a code_file
+    :arg someid: a string that's either a debug_id or a code_id
+    :arg refresh_cache: force a cache refresh
+
+    :returns: dict with keys from lookup_by_syminfo
+
+    """
+    key = f"{somefile}//{someid}"
```

Comment:
This key at least needs a prefix. It'd be good if it could use a version, too, so the cache is versioned across deploys. We could use get_version's commit.